app/products/calls/components/join_call_banner/index.ts

```diff
@@ -4,44 +4,60 @@
 import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
 import withObservables from '@nozbe/with-observables';
 import {of as of$} from 'rxjs';
-import {switchMap} from 'rxjs/operators';
+import {distinctUntilChanged, switchMap} from 'rxjs/operators';
 
 import JoinCallBanner from '@calls/components/join_call_banner/join_call_banner';
-import {observeCallsState, observeChannelsWithCalls, observeCurrentCall} from '@calls/state';
+import {observeCallsState, observeCurrentCall} from '@calls/state';
+import {idsAreEqual} from '@calls/utils';
 import {observeChannel} from '@queries/servers/channel';
-import {observeUsersById} from '@queries/servers/user';
+import {queryUsersById} from '@queries/servers/user';
 import {WithDatabaseArgs} from '@typings/database/database';
 
 type OwnProps = {
     serverUrl: string;
     channelId: string;
 }
 
-const enhanced = withObservables(['serverUrl', 'channelId'], ({serverUrl, channelId, database}: OwnProps & WithDatabaseArgs) => {
+const enhanced = withObservables(['serverUrl', 'channelId'], ({
+    serverUrl,
+    channelId,
+    database,
+}: OwnProps & WithDatabaseArgs) => {
     const displayName = observeChannel(database, channelId).pipe(
         switchMap((c) => of$(c?.displayName)),
+        distinctUntilChanged(),
     );
-    const currentCall = observeCurrentCall();
-    const participants = currentCall.pipe(
-        switchMap((call) => (call ? observeUsersById(database, Object.keys(call.participants)) : of$([]))),
+    const callsState = observeCallsState(serverUrl);
+    const participants = callsState.pipe(
+        switchMap((state) => of$(state.calls[channelId])),
+        distinctUntilChanged((prev, curr) => prev.participants === curr.participants), // Did the participants object ref change?
+        switchMap((call) => (call ? of$(Object.keys(call.participants)) : of$([]))),
+        distinctUntilChanged((prev, curr) => idsAreEqual(prev, curr)), // Continue only if we have a different set of participant ids
+        switchMap((ids) => (ids.length > 0 ? queryUsersById(database, ids).observeWithColumns(['lastPictureUpdate', 'last_picture_update']) : of$([]))),
```

Comment:
It should only need `'last_picture_update'`, since `'lastPictureUpdate'` shouldn't be a column in the table. Did you see any need to put both?

Reply:
Ah yes, thanks!